meson.build

```diff
@@ -0,0 +1,338 @@
+project('gtkhash',
+    'c',
+    version: '1.4',
+    meson_version: '>=0.53.0',
+    default_options: [
+        'warning_level=2',
+    ],
+)
+
+gnome       = import('gnome')
+pkgconfig   = import('pkgconfig')
+i18n        = import('i18n')
+
+################################################################################
+# Project configuration
+
+config = configuration_data()
+data_config = configuration_data()
+
+gettext_package = meson.project_name()
+config.set_quoted('GETTEXT_PACKAGE', gettext_package)
+
+config.set10('ENABLE_NLS', true)
+
+build_gtkhash   = get_option('build-gtkhash')
+enable_appstream= get_option('appstream')
+
+config.set10('ENABLE_NATIVE_FILE_CHOOSER', get_option('native-file-chooser'))
+
+config.set('HASH_FILE_BUFFER_SIZE', 131072, description: 'File read buffer size (bytes)')
+config.set('HASH_FILE_REPORT_INTERVAL', 166, description: 'Progress report interval (milliseconds)')
+
+
+config.set_quoted('VERSION',        meson.project_version())
+config.set_quoted('PACKAGE_NAME',   meson.project_name())
+config.set_quoted('PACKAGE',        meson.project_name())
+config.set_quoted('PACKAGE_STRING', meson.project_name() + ' ' + meson.project_version())
+config.set_quoted('G_LOG_DOMAIN',   meson.project_name())
+
+config.set_quoted('LOCALEDIR',      get_option('prefix') / get_option('localedir'))
+config.set_quoted('GNOMELOCALEDIR', get_option('prefix') / get_option('localedir'))
+
+data_config.set('VERSION', meson.project_version())
+data_config.set('RELEASE_DATE', '2020-07-11')
+
+################################################################################
+# Compiler options
+
+cc = meson.get_compiler('c')
+
+assert(get_option('warning_level').to_int() >= 2)
+
+# Todo check which options are used by Meson
+compiler_options = [
+    '-fno-common',
+    '-fvisibility=hidden',
+    '-pedantic',
+    '-Waggregate-return',
+    '-Wbad-function-cast',
+    '-Wcast-align',
+    '-Werror=implicit-function-declaration',
+    '-Werror=unknown-warning-option',
```

Comment:
```suggestion
```
meson seems to do this automatically (https://github.com/mesonbuild/meson/commit/b289547971f3bf11eae6da195ed4ed014aa57fdd)

Reply:
Yes, I was not 100% sore so I let it, but now that you pointed me to the code, I'll remove it :)